mongorestore/mongorestore_test.go

```diff
@@ -2345,3 +2346,223 @@ func runMongodump(t *testing.T, dir, db, collection string, args ...string) stri
 func uniqueDBName() string {
 	return fmt.Sprintf("mongorestore_test_%d_%d", os.Getpid(), time.Now().UnixMilli())
 }
+
+func TestRestoreColumnstoreIndex(t *testing.T) {
+	require := require.New(t)
+
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	session, err := testutil.GetBareSession()
+	require.NoError(err, "can connect to server")
+
+	fcv := testutil.GetFCV(session)
+	if cmp, err := testutil.CompareFCV(fcv, "6.3"); err != nil || cmp < 0 {
+		t.Skipf("Requires server with FCV 6.3 or later and we have %s", fcv)
+	}
+
+	t.Run("restore from dump", func(t *testing.T) {
+		testRestoreColumnstoreIndexFromDump(t)
+	})
+
+	res := session.Database("admin").RunCommand(context.Background(), bson.M{"replSetGetStatus": 1})
+	if res.Err() != nil {
+		t.Skip("server is not part of a replicaset so we cannot test restore from oplog")
+	}
+	t.Run("restore from oplog", func(t *testing.T) {
+		testRestoreColumnstoreIndexFromOplog(t)
+	})
+	//t.Run("restore from oplog with default index name", func(t *testing.T) {
+	//	testRestoreColumnstoreIndexFromOplog(t, "custom index name")
+	//})
+}
+
+func testRestoreColumnstoreIndexFromDump(t *testing.T) {
```

Comment:
I know there aren't comments on the old code which is where the structure of these functions and the test as a whole are from, but I think its important that we add robust comments here. Lets add docstrings to all functions you add, so its very easy to follow along with what is happening at each point.